feat: create hidden page for new agent auth product (#155)
langchain-ai/docs: src/langchain-agent-auth.mdx

---
title: LangChain Agent Auth
---
LangChain Agent Auth enables secure access from agents to any other system via OAuth 2.0 credentials. With LangChain Agent Auth, build your agents to securely connect to any system on a person's behalf.

## Overview

LangChain Agent Auth is designed to solve the challenge of secure authentication for autonomous agents. Instead of embedding credentials directly in agent code or requiring manual intervention for each API call, agents can use the OAuth flow to obtain time-limited access tokens for external services.

The system consists of:
- **Authentication Server**: Manages OAuth flows and stores credentials securely
- **Client SDK**: Python library for easy integration with agents

## Installation

Install directly from the wheel file:

For use with LangGraph:
```bash

> pip install https://docs.langchain.com/langchain_auth-0.1.0-py3-none-any.whl[langgraph]

Author: this will change once we host the package on PyPi in our typical way, but for now any alpha user can download the package from our docs site (where it is less visible to the public than PyPi)

#### In LangGraph Context

By default, tokens will be scoped to the calling agent using the Assistant ID parameter.

```python
auth_result = await client.authenticate(
    provider="github",
    scopes=["repo"],
    user_id="your_user_id" # Any unique identifier to scope this token to the human caller
)

# Or if you'd like a token that can be used by any agent, set agent_scoped=False
auth_result = await client.authenticate(
    provider="github",
    scopes=["repo"],
    user_id="your_user_id",
    agent_scoped=False
)
```

During execution, if authentication is required, the SDK will throw an [interrupt](https://langchain-ai.github.io/langgraph/how-tos/human_in_the_loop/add-human-in-the-loop/#pause-using-interrupt). The agent execution pauses and presents the OAuth URL to the user:

![LangGraph Studio interrupt showing OAuth URL](./images/langgraph-auth-interrupt.png)

After the user completes OAuth authentication and we receive the callback from the provider, they will see the auth success page:

![GitHub OAuth success page](./images/github-auth-success.png)

> The agent resumes execution from the point it left off. The token can now be used for any API calls. We store and refresh OAuth tokens on our side, meaning any future uses of this service and scope from this user and agent will not require an OAuth flow.

Reviewer: ```suggestion
The agent then resumes execution from the point it left off at, and the token can be used for any API calls. We store and refresh OAuth tokens on our side so that future uses of the service by either the user or agent don't require an OAuth flow.
```

That may not be the best suggestion but the last sentence was a little hard to follow so I tried to rework it a little. If what I suggested was technically inaccurate, please feel free to ignore or edit it.

```python
token = auth_result.token
```

#### Outside LangGraph Context

> Grab the auth_url and provide it to the user for out-of-band OAuth flow.

Reviewer: ```suggestion
Grab the `auth_url` and provide it to the user for out-of-band OAuth flows.
```
